tests/foreman/api/test_ansible.py

```diff
@@ -602,3 +602,106 @@ def test_negative_ansible_job_timeout_to_kill(
         assert [i['output'] for i in result if i['output'] == 'Exit status: 120']
+
+    @pytest.mark.tier2
+    @pytest.mark.no_containers
+    @pytest.mark.rhel_ver_match('8')
+    @pytest.mark.parametrize('setting_update', ['remote_execution_effective_user'], indirect=True)
+    def test_positive_ansible_job_privilege_escalation(
+        self,
+        target_sat,
+        rhel_contenthost,
+        module_org,
+        module_location,
+        module_ak_with_synced_repo,
+        setting_update,
+    ):
+        """Verify privilege escalation defined inside ansible playbook tasks is working
+        when executing the playbook via Ansible - Remote Execution
+
+        :id: 8c63fd1a-2121-4cce-9ec1-ae12817c9cc4
+
+        :steps:
+            1. Register a RHEL host to Satellite.
+            2. Setup a user on that host.
+            3. Create a playbook.
+            4. Set the SSH user to the created user, and unset the Effective user.
+            5. Run the playbook.
+
+        :expectedresults: In the playbook, created user is expected instead root user.
+
+        :BZ: 1955385
+
+        :customerscenario: true
+        """
+        playbook = '''
+            ---
+            - name: Test Play
+              hosts: all
+              gather_facts: false
+              tasks:
+                - name: Check current user
+                  command: bash -c "whoami"
+                  register: def_user
+                - debug:
+                    var: def_user.stdout
+                - name: Check become user
+                  command: bash -c "whoami"
+                  become: true
+                  become_user: testing
+                  register: bec_user
+                - debug:
+                    var: bec_user.stdout
+        '''
+        result = rhel_contenthost.register(
+            module_org, module_location, module_ak_with_synced_repo.name, target_sat
+        )
+        assert result.status == 0, f'Failed to register host: {result.stderr}'
+        assert rhel_contenthost.execute('useradd rexuser').status == 0
+        rex_passwd1 = rhel_contenthost.execute(
+            f'echo {settings.server.ssh_password} | passwd rexuser --stdin'
+        )
+        assert 'passwd: all authentication tokens updated successfully.' in rex_passwd1.stdout
+        pswd = rhel_contenthost.execute(
+            'echo "rexuser   ALL=NOPASSWD:   ALL" | tee -a /etc/sudoers.d/rexuser'
+        )
+        assert pswd.status == 0
+        assert rhel_contenthost.execute('useradd testing').status == 0
+        rex_passwd2 = rhel_contenthost.execute(
+            f'echo {settings.server.ssh_password} | passwd testing --stdin'
+        )
+        assert 'passwd: all authentication tokens updated successfully.' in rex_passwd2.stdout
+        rhel_contenthost.execute('su - testing -c "whoami"')
+        setting_update.value = ''
+        setting_update = setting_update.update({'value'})
+        assert setting_update.value == ''
```

Comment:
this looks redundant, can we remove this lines?

Reply:
I've removed redundant lines.